pdtf-api-1.2.0.yaml

```diff
@@ -0,0 +1,158 @@
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: https://raw.githubusercontent.com/Property-Data-Trust-Framework/schemas/master/src/schemas/pdtf-transaction.json
+        "404":
+          description: No claims are available for this property
+      description: Get a list of claims for a given transaction ID
+      operationId: getTransactionsById
+  /transactions/propertyPack:
+    parameters:
+      - $ref: "#/components/parameters/transactionIdInQueryParameter"
+      - $ref: "#/components/parameters/uprn"
+    get:
+      summary: Get latest state of data
+      tags:
+        - transactions
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: string
+                description: JSON representation of property pack information
+            text/html:
+              schema:
+                type: string
+                description: HTML renedered property pack information
+      operationId: getCurrentTransactionState
+      description: Retrieve a JSON or HTML representation of the state of the property data when all claims to date are applied.
+      parameters: []
+  /transactions/{id}/subscribe:
+    parameters:
+      - $ref: "#/components/parameters/transactionIdInPathParameter"
+    post:
+      summary: Create a new webhook subscription for this transaction
+      tags:
+        - transactions
+      operationId: transactionwebhookSubscribe
+      responses:
+        "201":
+          description: Created
+        "400":
+          description: Bad Request
+        "401":
+          description: Unauthorized
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                callbackUrl:
+                  type: string
+                  format: uri
+      callbacks:
+        newClaims:
+          "{$request.body#/callbackUrl}":
+            post:
+              requestBody:
+                required: true
+                content:
+                  application/json:
+                    schema:
+                      $ref: https://raw.githubusercontent.com/Property-Data-Trust-Framework/schemas/master/src/schemas/pdtf-verified-claims.json
+              responses:
+                "200":
+                  description: Please send a 200 response to acknowledge the callback and prevent retries
+    delete:
+      summary: Delete webhook
+      tags:
+        - transactions
+      operationId: transactionwebhookUnsubscribe
+      responses:
+        "200":
+          description: OK
+      description: Delete webhook for this UPRN for the current authenticated user, if any
+components:
+  parameters:
+    transactionIdInPathParameter:
+      required: true
+      name: id
+      in: path
+      schema:
+        type: string
```

Comment:
Should we formally require these to be `uuid`'s per [the spec](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.1.md#data-types)?

Reply:
I did try to specify `uuid`, but open-api spec does not support that type (https://swagger.io/docs/specification/data-models/data-types/)

Reply:
Ah; I now realize its the `format`; will add it!